packages/date/src/index.js

```diff
@@ -314,58 +315,197 @@ const formatMap = {
 		);
 	},
 	// Full date/time
-	c: 'YYYY-MM-DDTHH:mm:ssZ', // .toISOString
-	r: 'ddd, D MMM YYYY HH:mm:ss ZZ',
-	U: 'X',
+	c( dateValue ) {
+		return formatTZ( dateValue, "yyyy-MM-dd'T'HH:mm:ssxxx", {
+			timeZone: getActualTimezone(),
+		} );
+	},
+	r( dateValue ) {
+		return formatTZ( dateValue, 'EEE, dd MMM yyyy HH:mm:ss xx', {
+			timeZone: getActualTimezone(),
+		} );
+	},
+	U( dateValue ) {
+		return getUnixTime( utcToZonedTime( dateValue, getActualTimezone() ) );
```

Comment:
Could just use `t`?

Reply:
According to the docs, it seems the 't' format would work well https://date-fns.org/v1.30.1/docs/format.